### PR TITLE
CPLAT-3968 CPLAT-3969 Add unconvertJsProps() util to react_client.dart

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Description

The `getJsProps()` method in over_react makes more sense in this library, as it is intended to be used only with JS components and not Dart components. over_react has its own `getProps()` that should be sufficient (and uses `getJsProps()` internally).

To avoid naming conflicts and ambiguous imports in over_react, I've added the same functionality under a different name: `unconvertJsProps()`. This should also help clarify its intended usage in the future when some additional JS map work is taken on.

Additionally, logic was added to handle the unconverting of the JS event handlers (they are converted to their synthetic counterparts when added), which is necessary for Dart 2 compatibility.

## Testing
- [ ] CI passes
- [x] Verify this branch with over_react (on dart 1 and dart 2) as well as any other internal consumer large enough to instill adequate confidence

## Code Review
@greglittlefield-wf @aaronlademann-wf @kealjones-wk @corwinsheahan-wf

### CODE REVIEW

**Comment by greglittlefield-wf** [comment]
+1

File: lib/react_client.dart [hunk 1]
```diff
@@ -510,6 +511,28 @@ _convertBoundValues(Map args) {
 /// to the original Dart functions (the input of [_convertEventHandlers]).
 final Expando<Function> _originalEventHandlers = new Expando();
 
+/// Returns the props for a [ReactElement] or composite [ReactComponent] [instance],
+/// shallow-converted to a Dart Map for convenience.
```
Comment (greglittlefield-wf) [inline lib/react_client.dart]: #nit might be nice to mention the event handler unconverting

Comment (evanweible-wf) [reply lib/react_client.dart]: Added

**Comment by evanweible-wf** [comment]
@greglittlefield-wf I went ahead and added a `dartfmt` check to CI since we recently formatted this repo

**Comment by greglittlefield-wf** [comment]
+1

File: test/react_client_test.dart [hunk 2]
```diff
@@ -107,17 +107,46 @@ main() {
...
+        }
+      });
+
+      test('for a DOM element', () {
+        var component = react.div(props);
+        var jsProps = unconvertJsProps(component);
+        for (final key in eventPropKeyToEventFactory.keys) {
+          expect(jsProps[key], isNotNull,
+              reason: 'JS event handler prop should not be null');
+          expect(identical(jsProps[key], originalHandlers[key]), isTrue,
```
Comment (greglittlefield-wf) [inline test/react_client_test.dart]: FYI there's a `same` matcher you can use

**Comment by greglittlefield-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
Consumer tests:
- over_react (dart 1): https://travis-ci.org/Workiva/over_react/builds/490271779
- over_react (dart 2): https://travis-ci.org/Workiva/over_react/builds/490273413
- WSD: https://ci.webfilings.com/build/1620067

**Comment by greglittlefield-wf** [comment]
+1, merging

**Pull request merged**